source/dotnet/Library/AdaptiveCards.Templating/AdaptiveCardsTemplateVisitor.cs

```diff
@@ -282,15 +295,19 @@ public override AdaptiveCardsTemplateResult VisitTemplateRootData([NotNull] Adap
             }
             catch (ArgumentNullException e)
             {
-                throw new Exception($"'{context.GetText()}' at line, '{context.Start.Line}' is invalid because '{e.Message}'");
+                throw new AdaptiveTemplateException($"'{context.GetText()}' at line, '{context.Start.Line}' is invalid because '{e.Message}'");
             }
             catch (NullReferenceException)
             {
-                throw new Exception($"value of '$data : ', json pair, '{context.GetText()}'  at line, '{context.Start.Line}'is malformed");
+                throw new AdaptiveTemplateException($"value of '$data : ', json pair, '{context.GetText()}'  at line, '{context.Start.Line}'is malformed");
+            }
+            catch (JsonException e)
+            {
+                throw new AdaptiveTemplateException($"value of '$data : ', json pair, '{child.TEMPLATEROOT().Symbol.Text}' at line, '{child.TEMPLATEROOT().Symbol.Line}' is malformed", e);
             }
-            catch (Exception e)
+            catch
```

Comment:
Same: you can remove this block that is doing nothing but throwing the inner exception

Reply:
please see iteration 2.

---
In reply to: [426861321](https://github.com/microsoft/AdaptiveCards/pull/4026#discussion_r426861321) [](ancestors = 426861321)